targets/csharp/source/source/PlayFabGameserverInternal.cs

```diff
@@ -0,0 +1,324 @@
+            get { return _state; }
+            set
+            {
+                if (_state != value)
+                {
+                    _state = value;
+                    _signalHeartbeatEvent.Set();
+                }
+            }
+        }
+
+        public IList<GameserverConnectedPlayer> ConnectedPlayers { get; set; }
+
+        public IList<string> InitialPlayers { get; set; }
+
+        // Keep an instance of the callbacks around so they don't get garbage collected
+        public Action ShutdownCallback { get; set; }
+        public Func<bool> HealthCallback { get; set; }
+        public Action<DateTimeOffset> MaintenanceCallback { get; set; }
+
+
+        public GameserverInternalSdk(string configFileName = null)
+        {
+            _overrideConfigFileName = configFileName;
+            ConnectedPlayers = new List<GameserverConnectedPlayer>();
+            InitialPlayers = new List<string>();
+            TransitionToActiveEvent = new ManualResetEvent(false);
+        }
+
+        public Task StartAsync(bool debugLogs = false)
+        {
+            // If we already initialized everything, no need to do it again
+            if (_heartbeatTask != null)
+            {
+                return Task.CompletedTask;
+            }
+
+            _debug = debugLogs;
+
+
+            if (_configuration == null)
+            {
+                _configuration = GetConfiguration();
+            }
+
+            if (ConfigMap == null)
+            {
+                ConfigMap = CreateConfigMap(_configuration);
+            }
+
+            Logger = LoggerFactory.CreateInstance(ConfigMap[PlayFabGameserverSDK.LogFolderKey]);
+            if (_configuration.ShouldLog())
+            {
+                Logger.Start();
+            }
+
+            string gsmsBaseUrl = ConfigMap[PlayFabGameserverSDK.HeartbeatEndpointKey];
+            string instanceId = ConfigMap[PlayFabGameserverSDK.ServerIdKey];
+
+            Logger.Log($"VM Agent Endpoint: {gsmsBaseUrl}");
+            Logger.Log($"Instance Id: {instanceId}");
+
+            _webClient = GameserverHttpClientFactory.CreateInstance($"http://{gsmsBaseUrl}/v1/sessionHosts/{instanceId}");
+
+            _heartbeatRunning = true;
+            _signalHeartbeatEvent.Reset();
+            TransitionToActiveEvent.Reset();
+
+            _heartbeatTask = Task.Run(HeartbeatAsync);
+            return Task.CompletedTask;
+        }
+
+        private GameserverConfiguration GetConfiguration()
+        {
+            string fileName;
+
+            if (!string.IsNullOrWhiteSpace(_overrideConfigFileName))
+            {
+                fileName = _overrideConfigFileName;
+            }
+            else
+            {
+                fileName = Environment.GetEnvironmentVariable(PlayFabGameserverSDK.GsdkConfigFileEnvVarKey);
+            }
+
+            GameserverConfiguration localConfig;
+
+            if (!string.IsNullOrWhiteSpace(fileName) && File.Exists(fileName))
+            {
```

Comment:
>{ [](start = 12, length = 1)

we should add log messages here, but not sure if the logging is initialized at this point lol

Reply:
it hasn't been, since we check the config to see if logging is enabled, but I could write to console, what logging were you thinking?

---
In reply to: [255766683](https://github.com/PlayFab/SDKGenerator/pull/398#discussion_r255766683) [](ancestors = 255766683)

Reply:
ah, this won't be an issue anymore since I'm getting rid of the environment variable config

---
In reply to: [256058800](https://github.com/PlayFab/SDKGenerator/pull/398#discussion_r256058800) [](ancestors = 256058800,255766683)